test/topology_custom/test_tablets.py

```diff
@@ -324,3 +325,44 @@ async def test_read_of_pending_replica_during_migration(manager: ManagerClient,
 
     rows = await cql.run_async("SELECT pk from test.test")
     assert len(list(rows)) == 1
+
+
+# This test checks that --enable-tablets option and the TABLETS parameters of the CQL CREATE KEYSPACE
+# statemement are mutually correct from the "the least surprising behavior" concept. See comments inside
+# the test code for more details.
+@pytest.mark.parametrize("with_tablets", [True, False])
+@pytest.mark.parametrize("replication_strategy", ["NetworkTopologyStrategy", "SimpleStrategy", "EverywhereStrategy", "LocalStrategy" ])
+@pytest.mark.asyncio
+async def test_keyspace_creation(manager: ManagerClient, with_tablets, replication_strategy):
+    cfg = {'enable_user_defined_functions': False, 'enable_tablets': with_tablets}
+    server = await manager.server_add(config=cfg)
+    cql = manager.get_cql()
+
+    # Tablets are only possible when enabled and the replication strategy is NetworkTopology one
+    tablets_possible = (replication_strategy == 'NetworkTopologyStrategy') and with_tablets
+
+    # First, check if a kesypace is able to be created with default CQL statement that
+    # doesn't contain tablets parameters. When possible, tablets should be activated
+    await cql.run_async(f"CREATE KEYSPACE test_d WITH replication = {{'class': '{replication_strategy}', 'replication_factor': 1}};")
```

Comment:
Should `CREATE` also return a warning if tablets are turned on in the configuration, r.s. doesn't support them and we CREATE such a KS? I think we should at least change the scylla.yaml as here https://github.com/scylladb/scylladb/pull/20088/files#diff-918f761d7c0fe0a63cdeb6309e56193bc6770716e371ebaef050897e00fa933f and maybe also command line description of `enable_tablets` config option.

Reply:
Not return, but print into logs. Yes, this is good to have too. Let's postpone it to followup

Reply:
OK